File: app/src/main/java/com/ryanrvldo/moviecatalogue/data/source/remote/response/MovieResponse.java
package com.ryanrvldo.moviecatalogue.data.source.remote.response;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;
import com.ryanrvldo.moviecatalogue.data.source.model.Movie;

import java.util.ArrayList;

import lombok.Data;

@Data
public class MovieResponse {
    @SerializedName("page")
    @Expose
    private int page;

    @SerializedName("results")
    @Expose
    private ArrayList<Movie> movieItems;

    @SerializedName("total_pages")
    @Expose
    private int totalPages;

    @SerializedName("total_results")
    @Expose
    private int totalResults;
}

File: app/src/main/java/com/ryanrvldo/moviecatalogue/data/remote/response/MovieResponse.kt
package com.ryanrvldo.moviecatalogue.data.remote.response

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import com.ryanrvldo.moviecatalogue.data.model.Movie

data class MovieResponse(
    @Expose @SerializedName("page") val page: Int,
    @Expose @SerializedName("results") val movieItems: List<Movie>,
    @Expose @SerializedName("total_pages") val totalPages: Int,
    @Expose @SerializedName("total_results") val totalResults: Int
)